workflow/scripts/plot_network_connectedness.py

```diff
@@ -0,0 +1,85 @@
+"""
+Plotting tools for analysing network connectedness
+"""
+
+import sys
+from typing import Iterable, Tuple
+
+import pandas as pd
+import geopandas as gpd
+import networkx
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def network_components(edges: pd.DataFrame, nodes: pd.DataFrame) -> Tuple[pd.DataFrame, pd.DataFrame, list[set]]:
```

Comment:
I think this could use [snkit.network.add_component_ids](https://snkit.readthedocs.io/en/latest/_modules/snkit/network.html#add_component_ids) as an almost direct replacement

Reply:
Will do.